Admission controller

cezarybednarz opened the pull request.

zimnx commented:
Why new PR was opened? This one lacks history of comments, i can't check whether you addressed them or how.

cezarybednarz commented:
Link to previous pull request with history of comments: https://github.com/scylladb/scylla-operator-autoscaler/pull/17

cezarybednarz committed:
Change vendor folder for admission controller

zimnx commented on cmd/admission-controller/admission_controller.go:
```diff
@@ -0,0 +1,73 @@
+					Logger:                        logger,
+					ScyllaClient:                  client,
+					UpdaterServiceAccountUsername: updaterServiceAccountUsername,
+					ScaledResources:               []string{"cpu"},
```
accept as app parameter

zimnx commented on pkg/admission_controller/validatingwebhook_test.go:
```diff
@@ -0,0 +1,152 @@
+			cluster:         doubleRackWithChangedCPU,
+			oldCluster:      doubleRackCluster,
+			scas:            autoModeDoubleScaList,
+			scaledResources: []string{"Memory"},
```
`Memory` -> `memory`

zimnx reviewed (changes requested):
small nits

zimnx commented on config/default/admission_controller_webhook_patch.yaml:
```diff
@@ -0,0 +1,23 @@
+      containers:
+        - name: admission-controller
+          ports:
+            - containerPort: 443
```
9443

zimnx reviewed (commented):
rebase is needed and one comment

zimnx commented on config/webhook/service.yaml:
```diff
@@ -0,0 +1,11 @@
+spec:
+  ports:
+    - port: 443
+      targetPort: webhook
```
`webhook-server`

cezarybednarz committed:
Add admission controller

Admission controller is a admission webhook which denies all changes to scaled values (members, cpu requests and cpu limits)
